app/services/account_builder.rb

```diff
@@ -0,0 +1,185 @@
+# object that are named `#{AccountSubclass}Builder` (i.e. NufsAccountBuilder).
+# See `self.for()` for additional detail.
+#
+# Example usage:
+#   factory = AccountBuilder.for("NufsAccount")
+#   account = factory.new(...).build
+#
+class AccountBuilder
+
+  attr_reader :account, :account_type, :current_user, :facility, :owner_user,
+    :params, :params_key
+
+  # Initialize the instance variables.
+  #
+  # Possible `options` include:
+  #   - account_type - subclassed `Account` class name required for new/build
+  #   - account - subclassed `Account` object required for update
+  #   - facility - a facility or facility null object
+  #   - current_user - required
+  #   - owner_user - passed only for new/build account objects
+  #   - params - required
+  #
+  def initialize(options = {})
+    @account_type = valid_account_type!(options[:account], options[:account_type], options[:facility])
+    @facility = options[:facility]
+    @owner_user = options[:owner_user]
+    @current_user = options[:current_user]
+    @params_key = Account.config.account_type_to_param(@account_type)
+    @params = parse_params(options[:account], options[:params], @params_key)
+    @account = options[:account] || new_account(@account_type)
+  end
+
+  # Factory method that returns a subclassed `AccountBuilder` if one exists for
+  # the given account_type; otherwise returns `AccountBuilder` class (self).
+  # Be careful that `Object.const_get()` doesn't provide false-negatives for
+  # lazy-loaded builder classes. The `account_type` argument accepts an
+  # `Account` class object, class name string/symbol, or underscored class name
+  # string/symbol.
+  #
+  # Example usage:
+  #   AccountBuilder.for("NufsAccount") => NufsAccountBuilder
+  #   AccountBuilder.for("WithoutBuilderAccount") => AccountBuilder
+  #
+  def self.for(account_type)
+    klass = "#{account_type}_builder".classify
+    Object.const_defined?(klass) ? klass.constantize : self
+  end
+
+  # Builds and returns a new account subclassed `Account` object.
+  # Used when creating an account.
+  def build
+    @account = assign_params(@account, @params, @params_key)
+    @account = build_account_users(@account, @current_user, @owner_user)
+    @account = set_affiliate(@account, @params, @params_key)
+    @account = set_created_by(@account, @current_user)
+    @account = set_facility(@account, @facility)
+    @account
+  end
+
+  # Modifies and returns an existing account subclassed `Account` object.
+  # Used for updating an account.
+  def update
+    @account = assign_params(@account, @params, @params_key)
+    @account = set_affiliate(@account, @params, @params_key)
+    @account = set_updated_by(@account, @current_user)
+    @account
+  end
+
+  protected
+
+  # Returns hash of potentially modified params.
+  # Account can potentially be nil if a `new` form hasn't been submitted yet.
+  def parse_params(account, params, key)
```

Comment:
Maybe it's because nothing's overriding it yet, but I don't really get what this method is doing. The other two methods this call really don't do anything. I would kind of expect the return value of a method like this to be pretty much equivalent to `params[Account.config.account_type_to_param(@account_type)]`